NUnit Migration: Interface implementation methods incorrectly converted to async

## Description

When running the TUnit NUnit migration code fixer, methods that implement interface members with `void` return types are incorrectly converted to `async Task` methods. This breaks the interface implementation because the interface still expects `void`.

## Steps to Reproduce

1. Create a project with classes that implement interfaces (like `IInterceptor`, `IProxyGenerationHook`) and use NUnit assertions
2. Add TUnit packages  
3. Run `dotnet format analyzers --severity info --diagnostics TUNU0001`
4. Build the project

## Expected Behavior

Methods that implement interface members should NOT have their return type changed. The assertions should either:
1. Use synchronous assertion patterns
2. Be left unchanged for manual migration

## Actual Behavior

Interface implementation methods are converted to async, causing build errors:
```
error CS0738: 'ChangeProxyTargetInterceptor' does not implement interface member 
'IInterceptor.Intercept(IInvocation)'. 'ChangeProxyTargetInterceptor.Intercept(IInvocation)' 
cannot implement 'IInterceptor.Intercept(IInvocation)' because it does not have the 
matching return type of 'void'.
```

## Example

**Interface definition (not modified by migration):**
```csharp
public interface IInterceptor
{
    void Intercept(IInvocation invocation);
}
```

**Before migration:**
```csharp
public class ChangeProxyTargetInterceptor : IInterceptor
{
    public void Intercept(IInvocation invocation)
    {
        var targetAccessor = invocation.Proxy as IProxyTargetAccessor;
        Assert.IsNotNull(targetAccessor);
        targetAccessor.DynProxySetTarget(target);
        invocation.Proceed();
    }
}
```

**After migration (current - broken):**
```csharp
public class ChangeProxyTargetInterceptor : IInterceptor
{
    public async Task Intercept(IInvocation invocation)  // <-- Breaks interface implementation
    {
        var targetAccessor = invocation.Proxy as IProxyTargetAccessor;
        await Assert.That(targetAccessor).IsNotNull();
        targetAccessor.DynProxySetTarget(target);
        invocation.Proceed();
    }
}
```

**Expected after migration:**

Option 1 - Use synchronous pattern:
```csharp
public class ChangeProxyTargetInterceptor : IInterceptor
{
    public void Intercept(IInvocation invocation)
    {
        var targetAccessor = invocation.Proxy as IProxyTargetAccessor;
        Assert.That(targetAccessor).IsNotNull().Wait();  // or synchronous alternative
        targetAccessor.DynProxySetTarget(target);
        invocation.Proceed();
    }
}
```

Option 2 - Leave unchanged for manual migration:
```csharp
public class ChangeProxyTargetInterceptor : IInterceptor
{
    public void Intercept(IInvocation invocation)
    {
        var targetAccessor = invocation.Proxy as IProxyTargetAccessor;
        Assert.IsNotNull(targetAccessor);  // Left for manual migration
        targetAccessor.DynProxySetTarget(target);
        invocation.Proceed();
    }
}
```

## Detection Suggestion

The code fixer could detect interface implementations by checking:
1. If the containing class implements any interfaces
2. If the method signature matches an interface member
3. If so, don't convert to async

## Impact

This is a critical issue because it generates code that cannot compile.

## Environment

- TUnit version: 1.9.91
- .NET version: .NET 8.0
- OS: Windows

--- a/TUnit.Analyzers.CodeFixers/Base/AsyncMethodSignatureRewriter.cs
+++ b/TUnit.Analyzers.CodeFixers/Base/AsyncMethodSignatureRewriter.cs
@@ -10,6 +10,84 @@ namespace TUnit.Analyzers.CodeFixers.Base;
 /// </summary>
 public class AsyncMethodSignatureRewriter : CSharpSyntaxRewriter
 {
+    private readonly HashSet<string> _interfaceImplementingMethods;
+
+    public AsyncMethodSignatureRewriter() : this(new HashSet<string>())
+    {
+    }
+
+    public AsyncMethodSignatureRewriter(HashSet<string> interfaceImplementingMethods)
+    {
+        _interfaceImplementingMethods = interfaceImplementingMethods;
+    }
+
+    /// <summary>
+    /// Collects method signatures that implement interface members.
+    /// This should be called BEFORE syntax modifications while the semantic model is still valid.
+    /// </summary>
+    public static HashSet<string> CollectInterfaceImplementingMethods(
+        CompilationUnitSyntax compilationUnit,
+        SemanticModel semanticModel)
+    {
+        var methods = new HashSet<string>();
+
+        foreach (var methodDecl in compilationUnit.DescendantNodes().OfType<MethodDeclarationSyntax>())
+        {
+            // Check for explicit interface implementation syntax
+            if (methodDecl.ExplicitInterfaceSpecifier != null)
+            {
+                methods.Add(GetMethodKey(methodDecl));
+                continue;
+            }
+
+            var methodSymbol = semanticModel.GetDeclaredSymbol(methodDecl);
+            if (methodSymbol == null)
+            {
+                continue;
+            }
+
+            // Check if this method explicitly implements an interface
+            if (methodSymbol.ExplicitInterfaceImplementations.Length > 0)
+            {
+                methods.Add(GetMethodKey(methodDecl));
+                continue;
+            }
+
+            // Check if this method implicitly implements an interface member
+            var containingType = methodSymbol.ContainingType;
+            if (containingType != null)
+            {
+                foreach (var iface in containingType.AllInterfaces)
+                {
+                    foreach (var member in iface.GetMembers().OfType<IMethodSymbol>())
+                    {
+                        var impl = containingType.FindImplementationForInterfaceMember(member);
+                        if (SymbolEqualityComparer.Default.Equals(impl, methodSymbol))
+                        {
+                            methods.Add(GetMethodKey(methodDecl));
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        return methods;
+    }
+
+    /// <summary>
+    /// Gets a unique key for a method declaration based on its signature.
+    /// This key is stable across syntax tree modifications.
+    /// </summary>
+    private static string GetMethodKey(MethodDeclarationSyntax node)
+    {
+        // Build a key from class name, method name, and parameter types
+        var className = node.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault()?.Identifier.Text ?? "";
+        var methodName = node.Identifier.Text;
+        var parameters = string.Join(",", node.ParameterList.Parameters.Select(p => p.Type?.ToString() ?? ""));
+        return $"{className}.{methodName}({parameters})";
+    }
+
     public override SyntaxNode? VisitMethodDeclaration(MethodDeclarationSyntax node)
     {
         // First, visit children to ensure nested content is processed
@@ -29,6 +107,21 @@ public class AsyncMethodSignatureRewriter : CSharpSyntaxRewriter
             return node;
         }
 
+        // Skip methods with ref/out/in parameters (they can't be async)
+        if (node.ParameterList.Parameters.Any(p =>
+            p.Modifiers.Any(SyntaxKind.RefKeyword) ||
+            p.Modifiers.Any(SyntaxKind.OutKeyword) ||
+            p.Modifiers.Any(SyntaxKind.InKeyword)))
+        {
+            return node;
+        }
+
+        // Skip if method implements an interface member (changing return type would break the implementation)
+        if (ImplementsInterfaceMember(node))
+        {
+            return node;
+        }
+
         // Convert the return type
         var newReturnType = ConvertReturnType(node.ReturnType);
 
@@ -38,6 +131,19 @@ public class AsyncMethodSignatureRewriter : CSharpSyntaxRewriter
         return node
             .WithReturnType(newReturnType)
             .WithModifiers(newModifiers);
+    }
+
+    private bool ImplementsInterfaceMember(MethodDeclarationSyntax node)
+    {
+        // Check for explicit interface implementation syntax (IFoo.Method)
+        if (node.ExplicitInterfaceSpecifier != null)
+        {
+            return true;
+        }
+
+        // Check if this method was identified as an interface implementation
+        var key = GetMethodKey(node);
+        return _interfaceImplementingMethods.Contains(key);
     }
 
     private static TypeSyntax ConvertReturnType(TypeSyntax returnType)

--- a/TUnit.Analyzers.CodeFixers/Base/BaseMigrationCodeFixProvider.cs
+++ b/TUnit.Analyzers.CodeFixers/Base/BaseMigrationCodeFixProvider.cs
@@ -50,6 +50,11 @@ public abstract class BaseMigrationCodeFixProvider : CodeFixProvider
 
         try
         {
+            // IMPORTANT: Collect interface-implementing methods BEFORE any syntax modifications
+            // while the semantic model is still valid for the original syntax tree
+            var interfaceImplementingMethods = AsyncMethodSignatureRewriter.CollectInterfaceImplementingMethods(
+                compilationUnit, semanticModel);
+
             // Convert assertions FIRST (while semantic model still matches the syntax tree)
             var assertionRewriter = CreateAssertionRewriter(semanticModel, compilation);
             compilationUnit = (CompilationUnitSyntax)assertionRewriter.Visit(compilationUnit);
@@ -58,7 +63,8 @@ public abstract class BaseMigrationCodeFixProvider : CodeFixProvider
             compilationUnit = ApplyFrameworkSpecificConversions(compilationUnit, semanticModel, compilation);
 
             // Fix method signatures that now contain await but aren't marked async
-            var asyncSignatureRewriter = new AsyncMethodSignatureRewriter();
+            // Pass the collected interface methods to avoid converting interface implementations
+            var asyncSignatureRewriter = new AsyncMethodSignatureRewriter(interfaceImplementingMethods);
             compilationUnit = (CompilationUnitSyntax)asyncSignatureRewriter.Visit(compilationUnit);
 
             // Remove unnecessary base classes and interfaces

--- a/TUnit.Analyzers.Tests/NUnitMigrationAnalyzerTests.cs
+++ b/TUnit.Analyzers.Tests/NUnitMigrationAnalyzerTests.cs
@@ -2474,6 +2474,70 @@ public class NUnitMigrationAnalyzerTests
         );
     }
 
+    [Test]
+    public async Task NUnit_InterfaceImplementation_NotConvertedToAsync()
+    {
+        // Methods that implement interface members should NOT be converted to async
+        // because that would break the interface implementation contract.
+        // The interface method contains no NUnit assertions, so no await is added.
+        // Only the test method (which doesn't implement an interface) gets converted to async.
+        await CodeFixer.VerifyCodeFixAsync(
+            """
+                using NUnit.Framework;
+                using System.Threading.Tasks;
+
+                public interface ITestRunner
+                {
+                    void Run();
+                }
+
+                {|#0:public class MyClass|} : ITestRunner
+                {
+                    [Test]
+                    public void TestMethod()
+                    {
+                        Assert.That(true, Is.True);
+                    }
+
+                    public void Run()
+                    {
+                        // This implements ITestRunner.Run() and should stay void
+                        var x = 1;
+                    }
+                }
+                """,
+            Verifier.Diagnostic(Rules.NUnitMigration).WithLocation(0),
+            """
+                using System.Threading.Tasks;
+                using TUnit.Core;
+                using TUnit.Assertions;
+                using static TUnit.Assertions.Assert;
+                using TUnit.Assertions.Extensions;
+
+                public interface ITestRunner
+                {
+                    void Run();
+                }
+
+                public class MyClass : ITestRunner
+                {
+                    [Test]
+                    public async Task TestMethod()
+                    {
+                        await Assert.That(true).IsTrue();
+                    }
+
+                    public void Run()
+                    {
+                        // This implements ITestRunner.Run() and should stay void
+                        var x = 1;
+                    }
+                }
+                """,
+            ConfigureNUnitTest
+        );
+    }
+
     private static void ConfigureNUnitTest(Verifier.Test test)
     {
         test.TestState.AdditionalReferences.Add(typeof(NUnit.Framework.TestAttribute).Assembly);

